projects/knora-ui/src/lib/viewer/values/color-value/color-value.component.ts

```diff
@@ -0,0 +1,115 @@
+  templateUrl: './color-value.component.html',
+  styleUrls: ['./color-value.component.scss']
+})
+export class ColorValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+
+  @Input() displayValue?: ReadColorValue;
+
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+
+  customValidators = [Validators.pattern(CustomRegex.COLOR_REGEX)];
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder) {
+    super();
+  }
+
+  getInitValue(): string | null {
+    if (this.displayValue !== undefined) {
+      return this.displayValue.color;
+    } else {
+      return null;
+    }
+  }
+
+  ngOnInit() {
+    // initialize form control elements
+    this.valueFormControl = new FormControl(null);
+    this.commentFormControl = new FormControl(null);
+
+    // subscribe to any change on the comment and recheck validity
+    this.valueChangesSubscription = this.commentFormControl.valueChanges.subscribe(
+      data => {
+        this.valueFormControl.updateValueAndValidity();
+      }
+    );
+
+    this.form = this.fb.group({
+      colorValue: this.valueFormControl,
+      comment: this.commentFormControl
+    });
+
+    this.resetFormControl();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    this.resetFormControl();
+  }
+
+  // unsubscribe when the object is destroyed to prevent memory leaks
+  ngOnDestroy(): void {
+    this.unsubscribeFromValueChanges();
+  }
+
+  getNewValue(): CreateColorValue | false {
+    if (this.mode !== 'create' || !this.form.valid) {
+      return false;
+    }
+
+    const newColorValue = new CreateColorValue();
+
+    newColorValue.color = this.valueFormControl.value;
+
+    if (this.commentFormControl.value !== null && this.commentFormControl.value !== '') {
+      newColorValue.valueHasComment = this.commentFormControl.value;
+    }
+
+    return newColorValue;
+  }
+
+  getUpdatedValue(): UpdateColorValue | false {
+    if (this.mode !== 'update' || !this.form.valid) {
+      return false;
+    }
+
+    const updatedColorValue = new UpdateColorValue();
+
+    updatedColorValue.id = this.displayValue.id;
+
+    updatedColorValue.color = this.valueFormControl.value;
+
+    // add the submitted comment to updatedIntValue only if user has added a comment
+    if (this.commentFormControl.value !== null && this.commentFormControl.value !== '') {
+      updatedColorValue.valueHasComment = this.commentFormControl.value;
+    }
+
+    return updatedColorValue;
+  }
+
+  // update dynamically the background-color and the input label according to the picked color
+  onColorChanged(updatedValue: any) {
+    if (updatedValue) {
+      // background color of the input
+      this.displayValue.color = updatedValue;
+      // label
+      this.displayValue.strval = updatedValue;
+      // input inner text (not visible)
+      this.form.get('colorValue').setValue(updatedValue);
```

Comment:
also I do not understand why you have to set the new value in the form. You have a binding: `[formControlName]="'colorValue'"`. 

I do not understand why there is another binding: `[(colorPicker)]="displayValue.color"`

Reply:
The idea is that your input component gets its initial value from `displayValue`. Then the user can change the value. If the user saves the value, an update gets performed and the new value is written back to `displayValue`. If the user hits cancel, then the input component should be reset according to `displayValue`.

In your input component, you should only have a binding to the form. Then the input component should update the `FormControl` as the user manipulates the value. If that is not possible with a third party component, it has to be wrapped in a custom form field implementing the `ControlValueAccessor `.